README.rst

```diff
@@ -2,3 +2,121 @@ cdist manifestation
 ===================
 
 A helper for writing cdist manifests in Python.
+
+Motivation
+----------
+
+Writing complex manifests in shell can be painful and laborious. With Python and
+a proper abstraction layer, we can write manifests more easily.
+
+Usage
+-----
+
+First of all make sure your manifest is executable and contains
+``#!/usr/bin/env python`` shebang.
+
+Types
+`````
+
+To use type ``__directory``, import ``directory`` (without leading underscores)
+from ``cdist_manifestation.types`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+Every type is a function accepting a single positional argument, ``object_id``
+which is optional, and arbitrary amount of keyword arguments, which are the
+type parameters.
+
+So, one could create a cdist object of type ``__directory`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import directory
+
+    directory('/my/dir', mode=755, owner='me', parents=True)
+
+This will translate to ``__directory /my/dir --mode 755 --owner me --parents``.
+
+When you want to use the same parameter multiple times, just provide an iterable
+like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.types import user_groups
+
+    user_groups('me', group=['group1', 'group2'])
+
+This will translate to ``__user_groups me --group group1 --group group2``.
+
+To sum it up, these are the rules for using type parameters:
+
+- Underscores in parameter name will be implicitly converted to hyphens
+- Parameter values are implicitly converted to strings, except:
+    - Booleans, which simply denote a presence of a boolean parameter
+    - Iterables like lists and tuples, which denote presence of multiple
+      instances of the same parameter. Items of these iterables will be
+      converted to strings implicitly, disregarding these exceptions.
+
+Dependencies
+````````````
+
+Every type returns its dependency string which can be passed to ``require``
+context manager from ``cdist_manifestation.dependencies``. This function
+emulates behavior of ``require`` environmental variable.
+
+For example, if creating a file depends on creating its parent directory first,
+you can use ``require()`` like this:
+
+.. code-block:: python
+
+    from cdist_manifestation.dependencies import require
+    from cdist_manifestation.types import directory, file
+
+    my_dir = directory('/my/dir')
+
+    with require(my_dir):
+        file('/my/dir/file')
+
+This will translate to:
```

Comment:
I'm not sure if 'will translate to' is a good description of what happens -- we are not generating shell code. Wouldn't
"This is equivalent to:" be better?
(this comment applies to all occurrences of this phrase)

Reply:
Changed.